components/centraldashboard/public/components/main-page.js

```diff
@@ -303,36 +308,58 @@ export class MainPage extends utilitiesMixin(PolymerElement) {
         return !registrationFlow && workgroupStatusHasLoaded;
     }
 
-    /**
-     * Revert the sidebar index if the item clicked is an external link
-     * @param {int} curr
-     * @param {int} old
-     */
-    _revertSidebarIndexIfExternal(curr, old=0) {
-        if (curr <= this.menuLinks.length + 2) return;
-        this.sidebarItemIndex = old;
-    }
-
     /**
      * Tries to determine which menu link to activate based on the provided
      * path.
      * @param {string} path
+     * @param {string} hashPath
      */
-    _setActiveMenuLink(path) {
-        let menuLinkIndex = this.menuLinks
-            .findIndex((m) => path.startsWith(m.link));
-        if (menuLinkIndex >= 0) {
-            // Adds 1 since Overview is hard-coded
-            this.sidebarItemIndex = menuLinkIndex + 1;
-        } else {
-            menuLinkIndex = this.externalLinks
-                .findIndex((m) => path.startsWith(m.link));
-            if (menuLinkIndex >= 0) {
-                this.sidebarItemIndex = menuLinkIndex + this.menuLinks.length+1;
-            } else {
-                this.sidebarItemIndex = -1;
-            }
+    _setActiveMenuLink(path, hashPath) {
+        const htmlElements = this._clearActiveLink();
+        let matchPath = path;
+        if (hashPath && hashPath !== '') {
```

Comment:
the ```&& hasPath !== ''``` is redundant. It will not go into the if body when ```hashPath``` is `''`

Reply:
Oh you are right, thanks!